src/bm_sim/context.cpp

```diff
@@ -527,6 +527,20 @@ Context::read_counters(const std::string &counter_name, size_t idx,
   return (*counter_array)[idx].query_counter(bytes, packets);
 }
 
```

Comment:
Here, I decided to implement two new methods in context that allow the control plane to read and write to psa_counters since changing anything here would also affect simple switch.